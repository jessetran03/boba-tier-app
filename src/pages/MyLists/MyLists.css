.my-ratings h2 {
  margin-bottom: 30px;
}

.my-ratings h4 {
  margin-bottom: 8px;
}

.my-ratings select {
  margin: 0 15px;
}

.my-ratings form {
  margin-bottom: 40px;
}

.my-lists {
  margin: 10px 100px;
}

.my-lists ol {
  list-style-position: inside;
  margin: 10px;
  padding: 10px;
}

.my-lists h4 {
  margin: 5px;
}

.my-lists li {
  padding: 10px 10px;
  margin: 10px auto;
  border-style: solid;
  border-radius: 25px;
  max-width: 500px;
  background-color: white;
}