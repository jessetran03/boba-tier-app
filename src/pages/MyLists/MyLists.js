import React, { Component } from 'react'
import './MyLists.css'
import Rating from '../../components/Rating/Rating'
import config from '../../config'
import TokenService from '../../services/token-service'

export default class Rankings extends Component {
  state = {
    drinks: [],
    shops: [],
    shopId: 'all',
  }

  componentDidMount() {
    this.getShops()
    this.getUserDrinks()
  }

  onChange = e => {
    this.setState({ shopId: e.target.value })
  }

  handleFilter = e => {
    e.preventDefault();
    this.getUserDrinks();
  }

  getShops = () => {
    return fetch(`${config.API_ENDPOINT}/shops`, {
      method: 'GET',
    })
      .then(res =>
        (!res.ok)
          ? res.json().then(e => Promise.reject(e))
          : res.json()
      )
      .then((shops) => {
        this.setState({ shops })
      })
      .catch(error => {
        console.error({ error })
      })
  }

  getUserDrinks = () => {
    const shopId = this.state.shopId
    return fetch(`${config.API_ENDPOINT}/drinks/ratings/${shopId}`, {
      method: 'GET',
      headers: {
        'authorization': `bearer ${TokenService.getAuthToken()}`
      }
    })
      .then(res =>
        (!res.ok)
          ? res.json().then(e => Promise.reject(e))
          : res.json()
      )
      .then((drinks) => {
        this.setState({ drinks })
      })
      .catch(error => {
        console.error({ error })
      })
  }

  render() {
    const shops = this.state.shops
    const drinks = this.state.drinks
    return (
      <section className='my-ratings'>
        <h2>My Rated Drinks</h2>
        <form onSubmit={this.handleFilter}>
          <h4>Filter by store:</h4>
          <select onChange={this.onChange}>
            <option key='all' value='all'>All</option>
            {shops.map(shop =>
              <option key={shop.id} value={shop.id}>{shop.shop_name}</option>
            )}
          </select>
          <button type='submit'>Apply Filter</button>
        </form>
          <section className="my-lists">
            <h4>{this.state.currentShop}</h4>
            <ul>
              {drinks.length === 0 &&
                <p>You have not rated any drinks for this shop</p>
              }
              {drinks
                .sort((a, b) => a.rating < b.rating ? 1 : -1 || a.id < b.id ? 1 : -1)
                .map(drink =>
                  <li key={drink.id}>
                    <h4>{drink.drink_name}</h4>
                    <Rating rating={drink.rating} />
                    <p><i>{drink.shop_name}</i></p>
                  </li>
                )}
            </ul>
          </section>
      </section>
    )
  }
}