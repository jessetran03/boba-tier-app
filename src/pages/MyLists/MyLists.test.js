import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import MyLists from './MyLists'
import renderer from 'react-test-renderer'

it('renders MyLists without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(
    <BrowserRouter>
      <MyLists />
    </BrowserRouter>,
    div
  )
  ReactDOM.unmountComponentAtNode(div)
})

it('render MyLists without data', () => {
  const wrapper = renderer.create(
    <BrowserRouter>
      <MyLists />
    </BrowserRouter>
  )
    .toJSON();
  expect(wrapper).toMatchSnapshot()
})