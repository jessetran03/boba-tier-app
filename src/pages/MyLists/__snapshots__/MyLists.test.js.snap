// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render MyLists without data 1`] = `
<section
  className="my-ratings"
>
  <h2>
    My Rated Drinks
  </h2>
  <form
    onSubmit={[Function]}
  >
    <h4>
      Filter by store:
    </h4>
    <select
      onChange={[Function]}
    >
      <option
        value="all"
      >
        All
      </option>
    </select>
    <button
      type="submit"
    >
      Apply Filter
    </button>
  </form>
  <section
    className="my-lists"
  >
    <h4 />
    <ul>
      <p>
        You have not rated any drinks for this shop
      </p>
    </ul>
  </section>
</section>
`;
